stash_engine/app/models/stash_engine/resource.rb

```diff
@@ -147,6 +147,29 @@ def create_share
                                                    status: %w[published embargoed] })
     end
 
+    # complicated join & subquery that may be reused to get the last curation state for each resource
+    SUBQUERY_FOR_LATEST_CURATION = 'SELECT resource_id, max(id) as id FROM stash_engine_curation_activities GROUP BY resource_id'.freeze
+    JOIN_FOR_LATEST_CURATION = "INNER JOIN (#{SUBQUERY_FOR_LATEST_CURATION}) subq ON stash_engine_resources.id = subq.resource_id " \
+      'INNER JOIN stash_engine_curation_activities ON subq.id = stash_engine_curation_activities.id'.freeze
```

Comment:
Not necessary, but you can use the following syntax to make the above SQL a little easier to read:
```ruby
    SUBQUERY_FOR_LATEST_CURATION = <<~SQL
      SELECT resource_id, max(id) as id
      FROM stash_engine_curation_activities
      GROUP BY resource_id
    SQL
```